app/src/org/commcare/android/framework/CommCareActivityUIController.java

```diff
@@ -0,0 +1,16 @@
+package org.commcare.android.framework;
+
+/**
+ * Interface to be implemented by any class acting as a ui controller for a CommCareActivity
+ *
+ * IMPORTANT: Any CommCareActivity that uses a CommCareActivityUIController must override
```

Comment:
You should be able to make an interface called something like `WithUIController` that defines `initUIController` and `getUIController`. Then make the home activity, login activity, etc implement that interface. In the CommCareActivity you can then redefine `usesUIController()` to `return this instanceof WithUIController;` and make appropriate calls to the `WithUIController` interface with casts (`((WithUIController)this).getUIController()`). 

This seems _slightly_ messier with the casts and all, but enforces the implementation constraints that you have in the comments.